### DISCUSSION TITLE
Re-run config check on external changes

### DISCUSSION THREAD

**Discussion by siliconbrain** [kube-logging/logging-operator#1309] [General]
**Describe the bug**:
Config checks are only run when the config has changed, but validity of the config might depend on other, external properties too.

**Expected behaviour**:
A new config check should be run when external but relevant changes occur (e.g.: Fluentd image version changes, required service becomes available, etc.)

**Steps to reproduce the bug**:
E.g. change Fluentd image version.

/kind bug

**Comment by pepov** [comment]
I was thinking about this and I have to disagree. Changes loaded into the pod runtime should be the scope for the configcheck. Changes that require rollouts are out of scope in my opinion.
